src/components/utils/popover/popover-abstract.ts

```diff
@@ -0,0 +1,318 @@
+
+    this.nodes.popoverContainer.appendChild(this.nodes.items);
+
+    this.listeners.on(this.nodes.popoverContainer, 'click', (event: Event) => this.handleClick(event));
+
+    this.nodes.popover = Dom.make('div', [
+      css.popover,
+      this.params.class,
+    ]);
+
+    this.nodes.popover.appendChild(this.nodes.popoverContainer);
+  }
+
+  /**
+   * Adds search to the popover
+   */
+  private addSearch(): void {
+    this.search = new SearchInput({
+      items: this.items,
+      placeholder: this.messages.search,
+      onSearch: this.onSearch,
+    });
+
+    const searchElement = this.search.getElement();
+
+    if (searchElement === undefined) {
+      return;
+    }
+
+    searchElement.classList.add(css.search);
+
+    this.nodes.popoverContainer?.insertBefore(searchElement, this.nodes.popoverContainer.firstChild);
+  }
+
+  /**
+   * Adds custom html content to the popover
+   *
+   * @param content - html content to append
+   */
+  private addCustomContent(content: HTMLElement): void {
+    this.nodes.customContent = content;
+    this.nodes.customContent.classList.add(css.customContent);
+    this.nodes.popoverContainer?.insertBefore(content, this.nodes.popoverContainer.firstChild);
+  }
+
+  /**
+   * Handles clicks inside popover
+   *
+   * @param event - item to handle click of
+   */
+  private handleClick(event: Event): void {
+    const item = this.getTargetItem(event);
+
+    if (item === undefined) {
+      return;
+    }
+
+    if (item.isDisabled) {
+      return;
+    }
+
+    if (item.children.length > 0) {
+      this.handleShowingNestedItems(item);
+
+      return;
+    }
+
+    /** Cleanup other items state */
+    this.items.filter(x => x !== item).forEach(x => x.reset());
+
+    item.handleClick();
+
+    this.toggleItemActivenessIfNeeded(item);
+
+    if (item.closeOnActivate) {
+      this.hide();
+    }
+  }
+
+  /**
+   * Toggles nothing found message visibility
+   *
+   * @param isDisplayed - true if the message should be displayed
+   */
+  private toggleNothingFoundMessage(isDisplayed: boolean): void {
+    this.nodes.nothingFoundMessage?.classList.toggle(css.nothingFoundMessageDisplayed, isDisplayed);
+  }
+
+  /**
+   * Toggles custom content visibility
+   *
+   * @param isDisplayed - true if custom content should be displayed
+   */
+  private toggleCustomContent(isDisplayed: boolean): void {
+    this.nodes.customContent?.classList.toggle(css.customContentHidden, isDisplayed);
+  }
+
+  /**
+   * - Toggles item active state, if clicked popover item has property 'toggle' set to true.
+   *
+   * - Performs radiobutton-like behavior if the item has property 'toggle' set to string key.
+   * (All the other items with the same key get inactive, and the item gets active)
+   *
+   * @param clickedItem - popover item that was clicked
+   */
+  private toggleItemActivenessIfNeeded(clickedItem: PopoverItem): void {
+    if (clickedItem.toggle === true) {
+      clickedItem.toggleActive();
+    }
```

Comment:
huh?

Reply:
there should be return since if `clickedItem.toggle === true`, so check for `typeof clickedItem.toggle === 'string'` will never passed, that means that code below is not reachable. And its better to return here if it is not supposed to continue execution